src/DotLiquid/Util/StrFTime.cs

```diff
@@ -124,17 +150,28 @@ public static string ToStrFTime(this DateTime dateTime, string format, CultureIn
         /// <param name="culture">the CurrentCulture to be used when formatting</param>
         /// <returns>a string version of date-time matching pattern.</returns>
         public static string ToStrFTime(this DateTimeOffset dateTimeOffset, string format, CultureInfo culture)
+            => ObjectToStrFTime(dateTimeOffset, format, culture);
+
+        private static string ObjectToStrFTime(object source, string format, CultureInfo culture)
         {
-            culture = culture ?? throw new ArgumentException(message: "CultureInfo is mandatory", paramName: "culture");
-            return Regex.Replace(input: format, pattern: SPECIFIER_REGEX,
-                evaluator: specifier => SpecifierEvaluator(
-                    specifier: specifier.Groups[0].Value,
-                    flags: specifier.Groups[GROUP_FLAGS].Captures.Cast<Capture>().Select(capture => capture.Value).ToList(),
-                    width: specifier.Groups[GROUP_WIDTH].Captures.Cast<Capture>().Select(capture => (int?)Convert.ToInt32(capture.Value)).FirstOrDefault(),
-                    directive: specifier.Groups[GROUP_DIRECTIVE].Captures.Cast<Capture>().Select(capture => capture.Value).FirstOrDefault(),
-                    source: dateTimeOffset,
-                    culture: culture
-                    ));
+            culture = culture ?? throw new ArgumentException(message: "CultureInfo is mandatory", paramName: nameof(culture));
+
+            try
+            {
+                return Regex.Replace(input: format, pattern: SPECIFIER_REGEX,
+                    evaluator: specifier => SpecifierEvaluator(
+                        specifier: specifier.Groups[0].Value,
+                        flags: specifier.Groups[GROUP_FLAGS].Captures.Cast<Capture>().Select(capture => capture.Value).ToList(),
+                        width: specifier.Groups[GROUP_WIDTH].Captures.Cast<Capture>().Select(capture => (int?)Convert.ToInt32(capture.Value)).FirstOrDefault(),
+                        directive: specifier.Groups[GROUP_DIRECTIVE].Captures.Cast<Capture>().Select(capture => capture.Value).FirstOrDefault(),
+                        source: source,
+                        culture: culture
+                        ));
+            }
+            catch (FormatException ex) when (ex.Message == "Only")
+            {
+                throw new FormatException($"String '{format}' contains parts which are not specific to the {source.GetType().Name}.");

```

Comment:
Likely not implemented localization in the repo, so just noting but likely not actionable in this PR, exception messages should be localized. They should come from a resource file (i.e. *.resx) and then translations can be provided for each culture.

Reply:
Noted. There is localization in the project with Italian as well, but in recent PR we haven't been attentive to properly address.